README/MATH/Prediction_Market_Math.tex

```diff
@@ -0,0 +1,1379 @@
+
+\begin{itemize}
+    \item \textbf{Exchange, Transform, Load:} The net positions that every trader has on any given market are calculated through what is basically an Exchange, Transform and Load (ETL) pipeline. We Exchange the data in the form of transactions, Transform it through a series of steps as shown in the chart below, and finally we Load the net positions as an object.
+    \item \textbf{Works Anytime:} The ETL pipeline shown in the chart works at any point in the market's history, including after the resolution. Positions could be calculated for a market resolution at ay given percentage between 0 pc and 100 pc or at 0 (NO) or 100 (YES) pc resolutions.
+\end{itemize}
+
+
+\begin{tikzpicture}[
+    node distance=1cm,
+    every node/.style={rectangle, rounded corners, draw=black, align=center, font=\small, minimum width=3.5cm},
+    arrow/.style={->, thick}
+]
+
+% Nodes
+\node (calcProbs) [fill=blue!10] {Calculate market probabilities \\ (wpam.CalculateMarketProbabilitiesWPAM)};
+
+% Parallel nodes: Divide Shares and Course Payouts, placed closer together
+\node (divideShares) [below left=1cm and 0.5cm of calcProbs, fill=green!10] {Divide market pool shares \\ (dbpm.DivideUpMarketPoolSharesDBPM)};
+\node (coursePayouts) [below right=1cm and -6.25cm of calcProbs, fill=green!10] {Calculate course payouts \\ (dbpm.CalculateCoursePayoutsDBPM)};
+
+% Next node: Normalization Factors
+\node (normFactors) [below=3cm of calcProbs, fill=orange!10] {Calculate normalization factors \\ (dbpm.CalculateNormalizationFactorsDBPM)};
+
+% Scaled Payouts
+\node (scaledPayouts) [below=of normFactors, fill=orange!10] {Calculate scaled payouts \\ (dbpm.CalculateScaledPayoutsDBPM)};
+
+% Final Adjustments and Aggregation
+\node (adjustPayouts) [below=of scaledPayouts, fill=red!10] {Adjust payouts \\ (dbpm.AdjustPayoutsFromNewest)};
+\node (aggregatePayouts) [below=of adjustPayouts, fill=red!10] {Aggregate user payouts \\ (dbpm.AggregateUserPayoutsDBPM)};
+\node (netPositions) [below=of aggregatePayouts, fill=red!20] {Net aggregate market positions \\ (dbpm.NetAggregateMarketPositions)};
+\node (end) [below=of netPositions, fill=gray!20] {Return net positions};
+
+% Arrows
+\draw[arrow] (calcProbs) -- (divideShares);
+\draw[arrow] (calcProbs) -- (coursePayouts);
+\draw[arrow] (divideShares) -- (normFactors);
+\draw[arrow] (coursePayouts) -- (normFactors);
+\draw[arrow] (normFactors) -- (scaledPayouts);
+\draw[arrow] (scaledPayouts) -- (adjustPayouts);
+\draw[arrow] (adjustPayouts) -- (aggregatePayouts);
+\draw[arrow] (aggregatePayouts) -- (netPositions);
+\draw[arrow] (netPositions) -- (end);
+
+\end{tikzpicture}
+
+\newpage
+
+\section{First Transaction}
+
+\begin{itemize}
+    \item \textbf{Initial Investment:} Every market is assumed to have an initial investment of at least 1. This investment comes from a fee assessed to the market creator. This initial investment goes both in the numerator and denominator of the probability calculation.
```

Comment:
What does "assessed to the market creator" mean?

Reply:
A fee is charged to the person who created the market